src/transformers/models/dac/modeling_dac.py

```diff
@@ -0,0 +1,683 @@
+
+        Args:
+            input_values (`torch.Tensor of shape `(batch_size, 1, time_steps)`):
+                Input audio data to encode,
+            n_quantizers (int, *optional*):
+                Number of quantizers to use. If None, all quantizers are used. Default is None.
+            return_dict (`bool`, *optional*):
+                Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+        Returns:
+            loss (`torch.Tensor` of shape `(batch_size, 1, time_steps)`):
+                Loss from the encoder model, comprising the weighted combination of the commitment and codebook losses.
+            quantized_representation (`torch.Tensor` of shape `(batch_size, dimension, time_steps)`):
+                Quantized continuous representation of input.
+            audio_codes (`torch.Tensor` of shape `(batch_size, num_codebooks, time_steps)`):
+                Codebook indices for each codebook (quantized discrete representation of input).
+            projected_latents (`torch.Tensor` of shape `(batch_size, num_codebooks * dimension, time_steps)`):
+                Projected latents (continuous representation of input before quantization).
+        """
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+
+        quantized_representation = self.encoder(input_values)
+        quantized_representation, audio_codes, projected_latents, commitment_loss, codebook_loss = self.quantizer(
+            quantized_representation, n_quantizers
+        )
+
+        loss = self.config.commitment_loss_weight * commitment_loss + self.config.codebook_loss_weight * codebook_loss
+
+        if not return_dict:
+            return (loss, quantized_representation, audio_codes, projected_latents)
+
+        return DacEncoderOutput(loss, quantized_representation, audio_codes, projected_latents)
+
+    def decode(
+        self,
+        quantized_representation: Optional[torch.Tensor],
+        audio_codes: Optional[torch.Tensor] = None,
+        return_dict: Optional[bool] = None,
+    ):
+        """Decode given latent codes and return audio data
+
+        Args:
+            quantized_representation (torch.Tensor of shape `(batch_size, dimension, time_steps)`):
+                Quantized continuous representation of input.
+            audio_codes (`torch.Tensor` of shape `(batch_size, num_codebooks, time_steps)`, *optional*):
+                The codebook indices for each codebook, representing the quantized discrete
+                representation of the input. This parameter should be provided if you want
+                to decode directly from the audio codes (it will overwrite quantized_representation).
+            return_dict (`bool`, *optional*):
+                Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+        Returns:
+            audio_values (`torch.Tensor` of shape `(batch_size, input_length)`):
+                Decoded audio data.
```

Comment:
This is a bit misleading - it may return a `DacDecoderOutput` object which contains `audio_data`